components/public-api-server/pkg/identityprovider/cache.go

```diff
@@ -225,18 +249,23 @@ func (rc *RedisCache) Signer(ctx context.Context) (jose.Signer, error) {
 		err := rc.persistPublicKey(ctx, rc.current)
 		if err != nil {
 			log.WithField("keyID", rc.currentID).WithError(err).Error("cannot repersist public key")
-			return nil, err
+			return err
 		}
 	}
+	return nil
+}
 
-	return jose.NewSigner(jose.SigningKey{
-		Algorithm: jose.RS256,
-		Key:       rc.current,
-	}, &jose.SignerOptions{
-		ExtraHeaders: map[jose.HeaderKey]interface{}{
-			jose.HeaderKey("kid"): rc.currentID,
-		},
-	})
+func (rc *RedisCache) sync(ctx context.Context) {
```

Comment:
Would you be able to add a test for the sync?

I suspect to do this, you'll need to pull the frequency config outside of this as the test should ideally test that the sync happens twice, in a given iterval

Reply:
What you're testing is that the `sync` method gets periodically invoked, after the RedisCache is constructured.

So this means you need to simulate the following behaviours:
* Nothing in the cache, your value is written
* Something in the cache, you receive the value
* You need to setup your periods such that you can inspect the state in between two syncs.

In practice, probably setting the sync period to something like 3 seconds and checking contents of redis in between is probably the most sensible.